lib/faye/authentication/server_extension.rb

```diff
@@ -26,6 +26,7 @@ def incoming(message, callback)
               end
             debug("Authentication failed: #{message['error']}")
           end
+          message.delete('signature')
```

Comment:
is the signature present even if `Faye::Authentication.authentication_required?` is false ?

Reply:
I think I found the answer, the JS has the same `authentication_required` check and only adds the signature when needed, am I right ?

Reply:
> I think I found the answer, the JS has the same authentication_required check and only adds the signature when needed, am I right ?

Yup.

Basically, I don't want to mess with the payload if the extension is not supposed to be active.

Reply:
Okay make sense :+1:

Reply:
if it's not there will it raise?

Reply:
No.

Reply:
Thanks. Will be a good idea to verify after this if we still have signtures in outbound messages :)

Reply:
I checked it locally:
Before:
![Selection_438](https://user-images.githubusercontent.com/2705796/64703004-e64fdb00-d4ab-11e9-9887-279508e3d4b9.png)

After:
![Selection_437](https://user-images.githubusercontent.com/2705796/64703026-ee0f7f80-d4ab-11e9-805c-c4a2d06efc4d.png)

Reply:
Yes I mean if we find message with signature for which `Faye::Authentication.authentication_required?` is false